sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/SparkDateTimeUtils.scala

```diff
@@ -197,8 +197,8 @@ trait SparkDateTimeUtils {
     rebaseJulianToGregorianDays(julianDays)
   }
 
-  private val zoneInfoClassName = "sun.util.calendar.ZoneInfo"
-  private val getOffsetsByWallHandle = {
+  private lazy val zoneInfoClassName = "sun.util.calendar.ZoneInfo"
```

Comment:
Is it necessary to use `lazy` for this since `zoneInfoClassName` is just a `String`? 'Lazy' comes with the cost of locking.

Reply:
On the other hand, if `TimeZone.getDefault.getOffset(localMillis)` can be uniformly used to get `timeZoneOffset` in the `toJavaDate` function, then the use of the `MethodHandle` mechanism can be avoided here.

Reply:
> On the other hand, if TimeZone.getDefault.getOffset(localMillis) can be uniformly used to get timeZoneOffset in the toJavaDate function, then the use of the MethodHandle mechanism can be avoided here.

It seems that `ZoneInfo.getOffset` doesn't support daylight saving transitions?

Reply:
Not sure, but: 
1. Using `TimeZone.getDefault.getOffset(localMillis)`, all tests can pass, perhaps the unit test coverage is not comprehensive, but what would be the counterexample?
2. Looking at the inverse function `fromJavaDate` of `toJavaDate`, it does not take into account the scenario of `ZoneInfo`